### PR TITLE
feat: adaptative requests logging

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-292

### CODE REVIEW

File: mergify_engine/worker.py [hunk 1]
```diff
@@ -315,6 +315,7 @@ async def consume(self, stream_name: StreamNameType) -> None:
                         owner_id, owner_login, sub, client, redis_cache
                     )
                     pulls = await self._extract_pulls_from_stream(installation)
+                    client.set_requests_ratio(len(pulls))
```
Comment (jd) [inline mergify_engine/worker.py:318]: can this be 0 ? 🤔

Comment (sileht) [reply mergify_engine/worker.py:318]: no, in theory, since we can't be here without pulls, I will ensure we don't divide by 0

**Review by jd — APPROVED** [review]

**Comment by jd** [comment]
Semantic PR did not run.

**Comment by jd** [comment]
@Mergifyio update

**Pull request merged**